app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/model/Browser.kt

```diff
@@ -0,0 +1,236 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.model
+
+/**
+ * A model representing a supported browser.
+ */
+data class Browser(
+    val packageName: String,
+    val possibleUrlFieldIds: List<String>,
+    val urlExtractor: (String) -> String? = { it },
+) {
+    constructor(
+        packageName: String,
+        urlFieldId: String,
+        urlExtractor: (String) -> String? = { it },
+    ) : this(
+        packageName = packageName,
+        possibleUrlFieldIds = listOf(urlFieldId),
+        urlExtractor = urlExtractor,
+    )
+}
+
+/**
+ * Determines if the [String] receiver is a package name for a supported browser and returns that
+ * [Browser] if it is a match.
+ */
+fun String.getSupportedBrowser(): Browser? =
+    ACCESSIBILITY_SUPPORTED_BROWSERS.find { it.packageName == this@getSupportedBrowser }
+
+/**
+ * A list of supported browsers and the field ID used to find the url bar.
+ *
+ * This list should be kept in order and match the list of compatibility browsers in the
+ * autofill_service_configuration.xml.
+ */
+private val ACCESSIBILITY_SUPPORTED_BROWSERS = listOf(
```

Comment:
❓ just for curiosity, are the urlField values sourced from somewhere specific, are we getting those from the existing impl in the MAUI app?

Reply:
Yea, these are are pulled from Maui.